synapseclient/client.py

```diff
@@ -9465,6 +9465,37 @@ async def rest_delete_async(
             **kwargs,
         )
 
+    async def rest_get_paginated_async(
+        self, uri: str, limit: int = 20, offset: int = 0
+    ) -> AsyncGenerator[Dict[str, Any], None]:
+        """
+        Get paginated results asynchronously
+
+        Arguments:
+            uri:     A URI that returns paginated results
+            limit:   How many records should be returned per request
+            offset:  At what record offset from the first should iteration start
+
+        Returns:
+            A generator over paginated results
+
+        The limit parameter is set at 20 by default. Using a larger limit results in fewer calls to the service, but if
+        responses are large enough to be a burden on the service they may be truncated.
+        """
+        prev_num_results = sys.maxsize
+        while prev_num_results > 0:
+            params = {"offset": offset, "limit": limit}
+            page = await self.rest_get_async(
+                uri=uri,
+                params=params,
+            )
+            results = page["results"] if "results" in page else page["children"]
+            prev_num_results = len(results)
+
+            for result in results:
+                offset += 1
+                yield result
```

Comment:
This function is not needed, please revert this and swap to using this call which already exists:

https://github.com/Sage-Bionetworks/synapsePythonClient/blob/be5586ea46078f4ec5c70dce1b5d10bc9312505c/synapseclient/api/api_client.py#L64